crates/autopilot/src/domain/settlement/mod.rs

```diff
@@ -140,6 +141,75 @@ impl Settlement {
 
         Ok(Self { trades, auction_id })
     }
+
+    /// Build a settlement from a `competition::Solution`.
+    ///
+    /// Since `competition::Solution` does not contain JIT orders, the resulting
+    /// settlement might be a subset of it's onchain observable twin.
+    pub fn from_solution(
+        solution: &competition::Solution,
+        auction: &auction::Auction,
+        auction_id: auction::Id,
```

Comment:
We have a struct called `AuctionWithId` which may be appropriate here.

Reply:
True, but it would require unnecessary copy of the whole `Auction` object. I can do it if you think it's worth it.

Reply:
> True, but it would require unnecessary copy of the whole Auction object.

Really? Why does the auction exist without its ID in the runloop? Can we not move it into type AuctionWithId and then pass a reference to that around? This shouldn't require a clone.

As soon as an auction has an ID why do we still pass it around without?

Reply:
I believe we can just adjust the code in the run_loop that eagerly destructures the `AuctionWithId` to get around the clone.